### PR TITLE
DRAFT: Add extensions for use with kotlinx.serialization

### PR DESCRIPTION
**Pull request by russhwolf**
Looking for API feedback. The idea here is to enable arbitrary datatypes to be encoded inside a `Settings` instance.

Some questions:
1. Are there serialization edge-cases that should be added to the tests? One thing currently missing is a test of polymorphic serialization.
1. Is it a problem that the delegate APIs talk directly to the encoder/decoder rather than through a format? I think that means polymorphic serialization won't work on the delegate, but I don't know if that's an important use-case or not
1. Does the public API seem clean enough or are there adjustments I should make?

The CI build will fail. I think it's because of missing native platforms in kotlinx.serialization

### CODE REVIEW

**Comment by russhwolf** [comment]
For the sake of documenting it out loud, I should note that the versioning questions mentioned [here](https://github.com/russhwolf/multiplatform-settings/pull/37#issuecomment-667550151) apply to this PR as well

File: multiplatform-settings-serialization/build.gradle.kts [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+    "watchosArm32",
+    "watchosArm64",
+    "watchosX86"
+)
+
+kotlin {
+    sourceSets {
+        commonMain {
+            dependencies {
+                implementation(kotlin("stdlib-common"))
```
Comment (JavierSegoviaCordoba) [inline multiplatform-settings-serialization/build.gradle.kts:58]: Should be unnecessary if you update Kotlin to 1.4

Comment (russhwolf) [reply multiplatform-settings-serialization/build.gradle.kts:58]: Yeah this branch is a little old but these will be going away when it merges

**Comment by russhwolf** [comment]
This was released with 0.7
